packages/design-system/src/molecules/sidebarOption/index.tsx

```diff
@@ -0,0 +1,129 @@
+import { useState } from 'react'
+import { FlatList, Pressable, StyleSheet, View } from 'react-native'
+import { Icon } from '../../atoms'
+import { TypographyV2 } from '../../atoms/typographyV2'
+import { useThemeV2 } from '../../hooks'
+import { container, margin, padding } from '../../styles'
+import { SidebarOptionProps, OptionProps } from './types'
+
+const styles = StyleSheet.create({
+  mainOptionContainer: {
+    position: 'relative',
+  },
+  selectedIndicator: {
+    width: 4,
+    height: 16,
+    borderTopEndRadius: 2,
+    borderBottomEndRadius: 2,
+    position: 'absolute',
+    left: -20,
+  },
+})
+
+export const SidebarOption: React.FC<
+  SidebarOptionProps & { selectedOptionId?: string }
+> = ({
+  id,
+  namedIcon,
+  title,
+  nestedOptions,
+  onPress,
+  selectedOptionId = '',
+}) => {
+  const theme = useThemeV2()
+  const [showNestedOptions, setShowNestedOptions] = useState(false)
+  const handlePress = () => {
+    if (nestedOptions?.length) {
+      setShowNestedOptions(!showNestedOptions)
+      return
+    }
+    onPress?.()
+  }
+  const optionsId = [id].concat(nestedOptions?.map((option) => option.id) || [])
+  const isSelected = optionsId.includes(selectedOptionId)
```

Comment:
Let's refactor a bit to improve the readability & optimize a bit via using HashSet instead of an array

```suggestion
  const nestedOptionIds = nestedOptions?.map((option) => option.id) || []
  const optionIdsSet = new Set([id, ...nestedOptionIds]
  const hasSelectedOptionId = optionIdsSet.has(selectedOptionId)
```